File: client/src/models/LibraryViewModel.ts
import { z } from 'zod';
import AddressViewModel from './AddressViewModel';
import OpenHoursViewModel from './OpenHoursViewModel';

const LibraryViewModel = z.object({
  id: z.string().uuid(),
  name: z.string(),
  reservationTime: z.number().int().positive(),
  hireTime: z.number().int().positive(),
  emailAddress: z.string(),
  phoneNumber: z.string(),
  latitude: z.number(),
  longitude: z.number(),
  address: AddressViewModel,
  openHours: OpenHoursViewModel,
  isBookCurrentlyAvailable: z.boolean().optional(),
});

export default LibraryViewModel;
export type LibraryViewModelType = z.infer<typeof LibraryViewModel>;

File: Server/src/Server.Application/ViewModels/LibraryViewModel.cs
namespace Server.Application.ViewModels;

public class LibraryViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public int ReservationTime { get; set; }
    public int HireTime { get; set; }
    public string EmailAddress { get; set; }
    public string PhoneNumber { get; set; }
    public double Longitude { get; set; }
    public double Latitude { get; set; }
    public OpenHoursViewModel OpenHours { get; set; }
    public AddressViewModel Address { get; set; }
}
